mopidy/core/actor.py

```diff
@@ -136,6 +138,106 @@ def tags_changed(self, tags):
             self.playback._stream_title = title
             CoreListener.send('stream_title_changed', title=title)
 
+    def on_start(self):
+        logger.debug("core on_start")
+        try:
+            coverage = []
+            if self._config and 'restore_state' in self._config['core']:
+                amount = self._config['core']['restore_state']
+                if not amount or 'off' == amount:
+                    pass
+                elif 'volume' == amount:
+                    coverage = ['volume']
+                elif 'load' == amount:
+                    coverage = ['tracklist', 'mode', 'volume', 'history']
+                elif 'last' == amount:
+                    coverage = ['tracklist', 'mode', 'play-last', 'volume',
+                                'history']
+                elif 'play' == amount:
+                    coverage = ['tracklist', 'mode', 'play-always', 'volume',
+                                'history']
+                else:
+                    logger.warn('Unknown value for config '
+                                'core.restore_state: %s', amount)
+            if len(coverage):
+                self.load_state('persistent', coverage)
+        except Exception as e:
+            logger.warn('Unexpected error: %s', str(e))
+        pykka.ThreadingActor.on_start(self)
+
+    def on_stop(self):
+        logger.debug("core on_stop")
+        try:
+            if self._config and 'restore_state' in self._config['core']:
+                amount = self._config['core']['restore_state']
+                if amount and 'off' != amount:
+                    self.save_state('persistent')
+        except Exception as e:
+            logger.warn('on_stop: Unexpected error: %s', str(e))
+        pykka.ThreadingActor.on_stop(self)
+
+    def save_state(self, name):
+        """
+        Save current state to disk.
+
+        :param name: a name (for later use with :meth:`load_state`)
+        :type name: str
+        """
+        if not name:
+            raise TypeError('missing file name')
+
+        file_name = os.path.join(
+            self._config['core']['data_dir'], name)
+        file_name += '.state'
+        logger.info('Save state to %s', file_name)
+
+        data = {}
```

Comment:
Yes, I started with a CoreState, and reverted it. I couldn't use it as root entry.
 `data = CoreState()` did not work. 
 `data['core'] = CoreState()` will work. One type check instead of four.  I could change that.

Mopidy version in the file name, as data member or both? I read to the end and it is as data member. (set by the caller of save())